src/AvatarPair.tsx

```diff
@@ -12,33 +13,37 @@ const Wrapper = styled('div')`
   position: relative;
 `
 
-const childStyles = props => ({
+const childStyles = (props: CSSObject) => ({
```

Comment:
What's this `CSSObject`?

Reply:
This will be a long answer :)

TypeScript has a compiler setting called `noImplicitAny`. By default, variables (and parameters) with no type annotations, which cannot be inferred from other means, are implicitly typed as `any`. The `any` type is assignable to and from every other type, so it's basically the "anything goes" type. The `noImplicitAny` setting means that the only way to get an expression with an `any` type is to explicitly declare it. Variables and parameters without types are treated as errors since they cannot be any.

In this case, the `childStyles` variable is just a function that takes an object `props` and returns an object. But since that parameter `props` has no type annotation, the compiler gives it the `any` type implicitly, which is an error when `noImplicitAny` is enabled (and in our case it is).

So, to find out what CSSObject is, in VSCode we can use F12 to _go to definition_. That takes us to these 3 declarations:

```ts
export type CSSProperties = CSS.Properties<string | number>;
export type CSSPseudos = { [K in CSS.Pseudos]?: CSSObject };
export interface CSSObject extends CSSProperties, CSSPseudos {
    [key: string]: CSSObject | string | number | undefined;
}
```

Line by line...

```ts
export type CSSProperties = CSS.Properties<string | number>;
```
Declaring a type called CSSProperties. This is using the `Properties` interface from the csstype library. This library declares types that are relevant for CSS applications. In this case, the Properties type is the type of an object that optionally contains every possible CSS attribute and value. The `<string | number>` is a type parameter that they just use to mean that any string or number is a valid value for a length (for example, this would permit something like `borderWidth: 42` since `42` is a `number`). This line is just creating a shorthand/alias. Instead of typing `CSS.Properties<string | number>` everywhere, it's just `CSSProperties`.

```ts
export type CSSPseudos = { [K in CSS.Pseudos]?: CSSObject };
```
This type declaration is using a TypeScript feature called _[mapped types](http://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types)_. First, I F12'd on `Pseudos` to find out that it's just a union type of all the possible pseudo-selector strings, e.g. `":disabled"` and `":first-child"`. The syntax `[K in CSS.Pseudos]?: CSSObject` means _create a key for every item in the Pseudos union type, make it optional (`?`), and declare its type to be of `CSSObject`._

```ts
export interface CSSObject extends CSSProperties, CSSPseudos {
    [key: string]: CSSObject | string | number | undefined;
}
```
An interface is just an object type. When an interface `extends` another type, all of the keys from the extended types are copied into the new type you are declaring.

Finally, it's adding `[key: string]: <types>` to this CSSObject interface. The CSSObject type contains everything from CSSProperties and CSSPseudos, but now every key that hasn't already been declared by one of those inherited types is added in the body of the interface. You might ask why it was important to bother with CSSProperties and CSSPseudos. This is purely for editor autocomplete support. In other words, you could add any key you wanted to this object and it would compile, but all of the known keys will be able to be autocompleted.